src/main/java/org/elasticsearch/script/ScriptService.java

```diff
@@ -553,8 +557,8 @@ public static void writeTo(ScriptType scriptType, StreamOutput out) throws IOExc
         public final String lang;
         public final String script;
 
-        public CacheKey(String lang, String script) {
-            this.lang = lang;
+        public CacheKey(ScriptEngineService scriptEngineService, String script) {
```

Comment:
done, I did leave a newCacheKey static method though that takes the engine as argument, to make sure that the same lang/type is used consistently, lemme know what you think